app/code/core/Mage/Adminhtml/Block/Widget/Grid.php

```diff
@@ -970,29 +970,29 @@ public function _exportIterateCollection($callback, array $args)
         $originalCollection = $this->getCollection();
         $count = null;
         $page  = 1;
-        $lPage = null;
-        $break = false;
 
-        while ($break !== true) {
+        while (true) {
             $collection = clone $originalCollection;
             $collection->setPageSize($this->_exportPageSize);
             $collection->setCurPage($page);
             $collection->load();
-            if (is_null($count)) {
-                $count = $collection->getSize();
-                $lPage = $collection->getLastPageNumber();
-            }
-            if ($lPage == $page) {
-                $break = true;
+
+            $count = $collection->count();
+
+            if ($count < $this->_exportPageSize) {
+                break;
             }
-            $page ++;
+
+            $page++;
 
             foreach ($collection as $item) {
                 call_user_func_array(array($this, $callback), array_merge(array($item), $args));
             }
             $collection->clear();
             unset($collection);
         }
+
+        return $this;
```

Comment:
I added this return to match the phpdoc

Reply:
I'd go the other way with it. Make the PhpDoc match what's actually happening.

There's no good reason why anyone should use this as a fluid interface. Too much `return $this` in Magento

Reply:
Turned your note into code.